### DISCUSSION TITLE
Custom /etc/hosts file?

### DISCUSSION THREAD

**Discussion by dirk-olmes** [AdguardTeam/AdGuardHome#4344] [Q&A]
I'm running AdGuardHome as a docker container. Since it's in a docker container I cannot modify the `/etc/hosts` file to serve the hostnames I use in my local network.

Is there a way to provide AGH with a different file in `/etc/hosts` format?

**Comment by ainar-g** [comment]
Sorry about the late response.  You can just add that file as a filter list on the *Filters → DNS blocklists* page.

**Reply by dirk-olmes** [reply]
I think this is not what I want. I don't wand to add `/etc/hosts` as a block list but rather as a file for additional DNS records.

Is that possible at all?

**Reply by ainar-g** [reply]
@dirk-olmes, a better way to do that is through the [`dnsrewrite` rules][1].

[1]: https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#dnsrewrite

**Reply by maajix** [reply]
I am having the same issue. The thing is that the rewriting does not work since clients connect from the router. So, then the container gets asked what is xyz.local and does not know the answer obviously since its a local ip that cant be resolved. For example if i directly use the Adguard DNS server (by hardcoding the ip for example) it works fine but when i want to use any other devices which first connects to my router it wont work..

**Reply by gabe565** [reply]
@LetsFailYourGame Maybe not a perfect solution, but I've had success with setting my AdGuard Home instance to forward my local domain name back to my router. In _Settings → DNS Settings → Upstream DNS Servers_, I added the following:

```
[/home/local/]192.168.1.1
```

In my case, my local domain name is `.home`, and I added `.local` for mDNS.

The only issue I've seen is the possibility of creating a never ending DNS loop. My ASUS router has a setting to disallow forwarding of local device queries to upstream DNS, so that prevents any possibility of looping, but I'm not sure if how standard an option like that is.

Hopefully that helps!
